internal/cluster/cluster.go

```diff
@@ -1039,6 +1040,88 @@ func (m *Manager) IsReadyForInstallation(c *common.Cluster) (bool, string) {
 	return true, ""
 }
 
+func (m *Manager) detectAndStoreCollidingIPsForCluster(cluster *common.Cluster, db *gorm.DB) error {
+	if db == nil {
+		db = m.db
+	}
+	// We want to calculate ip collisions only when in pre-install states since it is needed for pre-install validations
+	allowedStates := []string{
+		models.ClusterStatusPendingForInput,
+		models.ClusterStatusInsufficient,
+
+		models.ClusterStatusReady,
+	}
+	if !funk.ContainsString(allowedStates, swag.StringValue(cluster.Status)) {
+		return nil
+	}
+
+	hosts := cluster.Hosts
```

Comment:
Maybe copy to a new slice without changing the original slice ?

Reply:
@ori-amizur there are quite a few places in the code where the hosts are sorted in memory just so the order will be predictable. I believe it would be much faster to read the hosts from the db ORDER BY the hostId and have them already sorted on the cluster object. WDYT?

Reply:
I am not sure all places need the same order

Reply:
Most of them do